src/gengowatcher/web.py

```
@@ -858,6 +1140,70 @@ async def execute_command(
+    if not api_instance:
+        raise HTTPException(status_code=503, detail="API not initialized")
+    try:
+        return api_instance.list_files()
+    except Exception as e:
+        api_instance.logger.exception(f"Error listing files: {e}")
+        raise HTTPException(status_code=500, detail="Internal server error")
+
+
+@app.post("/api/files/upload", response_model=StoredFileUploadResponse)
+async def upload_file(
+    file: UploadFile = File(...),
```

Ruff (0.15.9)

[warning] Do not perform function call `File` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)